pkg/binlog/reader/tcp.go

```diff
@@ -114,6 +114,16 @@ func (r *TCPReader) Close() error {
 		return errors.New("already closed")
 	}
 
+	setStageClosed := func() error {
+		r.stage = stageClosed
+		return nil
+	}
+
+	if r.syncer == nil {
```

Comment:
when `r.syncer == nil`?

Reply:
remove this check. If we use `NewTCPReader` to get a TCPReader, `syncer` is always created